components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java

```diff
@@ -95,19 +99,21 @@ public String jwks() {
             while (enumeration.hasMoreElements()) {
                 String alias = (String) enumeration.nextElement();
                 if (keystore.isKeyEntry(alias)) {
-                    Certificate cert = keystore.getCertificate(alias);
-                    certificatesWithAliases.put(alias, cert);
+                    CertificateChainInfo certificateChainInfo = new CertificateChainInfo();
+                    certificateChainInfo.setCertificate(keystore.getCertificate(alias));
+                    certificateChainInfo.setCertificateChain(keystore.getCertificateChain(alias));
+                    certChainInfosWithAliases.put(alias, certificateChainInfo);
                 }
             }
-            return buildResponse(certificatesWithAliases);
+            return buildResponse(certChainInfosWithAliases);
         } catch (Exception e) {
             String errorMessage = "Error while generating the keyset for tenant domain: " + tenantDomain;
             return logAndReturnError(errorMessage, e);
         }
     }
 
-    private String buildResponse(Map<String, Certificate> certificates)
-            throws IdentityOAuth2Exception, ParseException {
+    private String buildResponse(Map<String, CertificateChainInfo> certChainInfosWithAliases)
```

Comment:
`certChainInfosWithAliases` ->` certtificateInfo`  should be fine.

Reply:
if the `certChainInfosWithAliases` constains alias, we don't need this map object, We can use a list of `certtificateInfo`